faas-samples/current-queue-health/README.md

```diff
@@ -0,0 +1,173 @@
+# Current Queue Health Cache
+
+The purpose of this function is to access data from the Current Queue Health API and store that data within the Conversation Context Service. Reporting APIs such as Current Queue Health are not intended for use at a conversational scale. As opposed to accessing the APIs directly from a Conversation Builder bot, this function will cache the data from this API. That data can then be accessed for routing and messaging decisions using bot context methods to access the Conversation Context Service.
+
+This process was originally documented on our [developers forum](https://talkyard.livepersonai.com/-18/caching-current-queue-health-apis-w-faas-the-context-session-store).
+
+---
+
+LivePerson provides a number of APIs which are designed to be used with Reporting Dashboards to inform on items such as the status of the queue and the shift status of agents. Having this information readily accessible within Conversation Builder can assist with correctly setting expectations or properly routing individuals when a skill may not be available to escalate to. Unfortunately, many of these reporting dashboard APIs are not intended for use at a Conversational Scale, and doing so could be detrimental to the stability of our platform.
+
+This guide is intended to provide an alternative to calling these APIs at every conversation. By using several Conversational Cloud services in conjunction with one another to regularly cache needed data, we can make this information accessible in a more robust service that is properly scaled to handle the volume of calls from each conversation.
```

Comment:
I wonder if we have any guidance to offer on managing these trade-offs. 

Like, if they have minimal volume (less than 100 chats per day or something), would all of this be overkill and they can just query the queue health API directly? Or do we advise that, for any client, if they want to use the queue health or shift status in a bot, they should cache it like this?

Reply:
All guidance I've received on this matter is that this is the proper way to get this information into the bots. While there are definitely brands out there accessing the information from the bot directly, we've been advised to promote this process as a best practice.